qtgui/resources/ApplicationWindow.qml

```diff
@@ -0,0 +1,76 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+import starturtle.oneBit 1.0
+
+ApplicationWindow {
+    id: mainWindow
+    visible: true
+    width: 800
+    height: 600
+    title: qsTr("Image Pixelation")
+
+    menuBar: MenuBar {
+        Menu {
+            title: qsTr("File")
+            MenuItem {
+                text: qsTr("&Load")
+                onTriggered: {
+					imagePreview.getInputFile()
+				}
+            }
+            MenuItem {
+                text: qsTr("Select Out&path")
+                onTriggered: {
+					imagePreview.getOutputFile()
+				}
+            }
+            MenuItem {
+                text: qsTr("Exit")
+                onTriggered: Qt.quit();
+            }
+        }
+		Menu {
+			title: qsTr("&Pixelation")
+			MenuItem {
+				text: qsTr("&Run")
+				onTriggered: {
+					pixelator.setInputImage(imagePreview.sourcePath)
+					pixelator.setOutputImage(imagePreview.storagePath)
+					pixelator.setStitchSizes(pixelSizes.resultWidth, pixelSizes.resultHeight, pixelSizes.stitchRows, pixelSizes.stitchColumns)
+					pixelator.setStitchColors(pixelColors.color1, pixelColors.color2)
+					var result = pixelator.run()
+					imagePreview.updateResult()
+				}
+			}
+		}
+    }
+
+	GridLayout
+	{
+		columns: 2
+		anchors.fill: parent
+		PixelSizes {
+			Layout.fillWidth: true
+			id: pixelSizes
+		}
+	
+		PixelColors {
+			id: pixelColors
+			Layout.fillWidth: true

```

Comment:
layout isn't quite right. the two panels differ in size, so there are two different frames. Either make them scale to same size, or remove the frame & set anchoring to "top" so it won't show...